File: Java/Shapes.java
// Grant Commodore

import java.util.Scanner;

/* This program computes the appropriate area, and/or perimeter and circumference.  
It also echoes any dimensions used.*/
public class Shapes {
    static Scanner input = new Scanner(System.in);

    // Get the user's selection from the menu.
    public static int getSelection() {
        System.out.println("|_______________________________________________|");
        System.out.println("| 1. Rectangle                                  |");
        System.out.println("| 2. Circle                                     |");
        System.out.println("| 3. Hexagon                                    |");
        System.out.println("| 4. Pentagon                                   |");
        System.out.println("| 5. Quit                                       |");
        System.out.println("|_______________________________________________|");
        System.out.println("Enter your selection: ");
        int selection = input.nextInt();
        return selection;
    }

    // Get the rectangle's length and width and calculate the area and perimeter.
    public static void getRectangle() {
        System.out.println("Enter the rectangle length: ");
        double length = input.nextDouble();
        System.out.println("Enter the rectangle width: ");
        double width = input.nextDouble();
        double area = length * width;
        double perimeter = 2 * (length + width);
        System.out.println("The area of the rectangle is: " + area);
        System.out.println("The perimeter of the rectangle is: " + perimeter);
    }

    // Gets the circle's radius and calculates the area and circumference.
    public static void getCircle() {
        System.out.println("Enter the circle's radius: ");
        double radius = input.nextDouble();
        double area = Math.PI * Math.pow(radius, 2);
        double circumference = 2 * Math.PI * radius;
        System.out.println("The area of the circle is: " + area);
        System.out.println("The circumference of the circle is: " + circumference);
    }

    // Gets the hexagon's side and calculates the area and perimeter.
    public static void getHexagon() {
        System.out.println("Enter the hexagon's side: ");
        double side = input.nextDouble();
        double area = (3 * Math.sqrt(3) / 2) * Math.pow(side, 2);
        double perimeter = 6 * side;
        System.out.println("The area of the hexagon is: " + area);
        System.out.println("The perimeter of the hexagon is: " + perimeter);
    }

    // Get the pentagon's side and calculates the area and perimeter.
    public static void getPentagon() {
        System.out.println("Enter the pentagon's side: ");
        double side = input.nextDouble();
        double area = (5 * Math.pow(side, 2)) / (4 * Math.tan(Math.PI / 5));
        double perimeter = 5 * side;
        System.out.println("The area of the pentagon is: " + area);
        System.out.println("The perimeter of the pentagon is: " + perimeter);
    }

    // Main method that begins the execution of the Java application.
    public static void main(String[] args) {
        // Run this program until the user selects the quit option.
        int selection = 0;
        while (selection != 5) {
            selection = getSelection();
            switch (selection) {
                case 1:
                    getRectangle();
                    break;
                case 2:
                    getCircle();
                    break;
                case 3:
                    getHexagon();
                    break;
                case 4:
                    getPentagon();
                    break;
                case 5:
                    System.out.println("Program Terminated!");
                    break;
                default:
                    System.out.println("Invalid selection. Please try again.");
                    break;
            }
        }

        input.close();
    }
}

File: php/shapes.php
<?php
// Grant Commodore
// This program computes the appropriate area, and/or perimeter and circumference.  
// It also echoes any dimensions used.

// Run the program until the selection is 0.
function menu () {
    echo "---------------------\n";
    echo "1: Hexagon \n";
    echo "2: Circle\n";
    echo "3: Pentagon\n";
    echo "4: Rectangle\n"; 
    echo "0: Quit\n";
    echo "---------------------\n";
    $selection = readline("Enter your selection: ");
    return $selection;
}

$selection = menu();

while ($selection != 0) {
    // Get the dimensions from the user based on the selected shape.
    if ($selection == 1) {
        // Get the dimensions for a hexagon.
        $side = readline("Enter the length of a side on the hexagon: ");
        $area = 3 * sqrt(3) / 2 * ($side * $side);
        $perimeter = 6 * $side;
        echo "The area of the hexagon is $area and the perimeter is $perimeter \n";
    } else if ($selection == 2) {
        // Get the dimensions for a circle.
        $radius = readline("Enter the radius of the circle: ");
        $area = pi() * $radius * $radius;
        $circumference = 2 * pi() * $radius;
        echo "The area of the circle is $area and the circumference is $circumference \n";
    } else if ($selection == 3) {
        // Get the dimensions for a pentagon.
        $side = readline("Enter the length of a side on the pentagon: ");
        $area = 5 * $side * $side / (4 * tan(pi() / 5));
        $perimeter = 5 * $side;
        echo "The area of the pentagon is $area and the perimeter is $perimeter \n";
    } else if ($selection == 4) {
        // Get the dimensions for a rectangle.
        $length = readline("Enter the rectangle length: ");
        $width = readline("Enter the rectangle width: ");
        $area = $length * $width;
        $perimeter = 2 * ($length + $width);
        echo "The area of the rectangle is $area and the perimeter is $perimeter \n";
    } else {
        // If the selection is not 1, 2, 3, 4, or 0, echo an error message.
        echo "Invalid selection \n";
    }
    // Get the next selection.
    echo "\n";
    $selection = menu();
}

echo "Program terminated \n";

?>